### ISSUE TITLE
Constuctor doesn't work the same as a real DataView?

### ISSUE THREAD

**Issue by greggman** [jDataView/jDataView#84]
AFAICT a real `DataView` requires an `ArrayBuffer` in it's constructor but a JDataView requires a `Buffer` or `Uint8Array`

Trying this in the browser console

      new DataView(new Uint8Array(3)); 

In Chrome I get `Uncaught TypeError: First argument to DataView constructor must be an ArrayBuffer` and in Firefox I get `TypeError: DataView: expected ArrayBuffer, got Uint8Array`.

If I use

      new DataView(new Uint8Array(3).buffer);

Then it works in both.

Conversely with JDataView 2.5.0 the first one succeeds even though it shouldn't according to the spec. The second one fails.

Is that what you want to happen?

**Comment by RReverser** [comment]
Yes, the entire point of jDataView is to support any platforms including those that don't have ArrayBuffer. It's a superset of DataView APIs, and so supports more sources.

**Comment by greggman** [comment]
You missed the point entirely. It's not a problem that it takes more sources. **IT'S A PROBLEM THAT IT DOESN'T WORK WITH SOURCES THAT DATAVIEW WORKS WITH**

In other words I have code like this

     xhr = new XMLHttpRequest();
     xhr.open('GET', ....)
     xhr.responseType = "arraybuffer";
     xhr.onload = (e) {
         dv = new DataView(xhr.response);  // xhr.response is an ArrayBuffer
     }
     xhr.send();

This works

Now I try to use `JDataView` isntead

     global.DataView = require('jdataview');

     xhr = new XMLHttpRequest();
     xhr.open('GET', ....)
     xhr.responseType = "arraybuffer";
     xhr.onload = (e) {
         dv = new DataView(xhr.response); // ****ERROR!!! JDataView doesn't work here
     }
     xhr.send();

I thought your goal was to be a drop in replacement for `DataView` with added features but as it is it's **NOT** because `JDataView` doesn't accept an `ArrayBuffer` as input whereas `DataView` does

**Comment by RReverser** [comment]
A bit easier on the tone, please ;)

> JDataView doesn't accept an ArrayBuffer as input whereas DataView does

That does sound as a bug, because jDataView should accept ArrayBuffer just fine.

Can you please provide a live example e.g. with [JS Bin](https://jsbin.com/?html,output) that would demonstrate the issue reproducibly?

Thanks.

**Comment by greggman** [comment]
Just follow these steps

```
$ npm install jdataview
$ node
global.DataView = require('jdataview');
const a = new DataView(new Uint8Array(3));  // works
const b = new DataView(new ArrayBuffer(3));  // fails
```

Here's my results

```
$ npm install jdataview
/Users/gregg/temp/delme-jdataview
└── jdataview@2.5.0 

$ node
> global.DataView = require('jdataview');
{ [Function: jDataView]
  wrapBuffer: [Function],
  is: [Function],
  from: [Function],
  Uint64: { [Function: Uint64] fromNumber: [Function] },
  Int64: { [Function: Int64] fromNumber: [Function] } }
> const a = new DataView(new Uint8Array(3));
undefined
> const b = new DataView(new ArrayBuffer(3));
RangeError: Maximum call stack size exceeded
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
    at Object.jDataView (/Users/gregg/temp/delme-jdataview/node_modules/jdataview/dist/node/jdataview.js:27:77)
```

Try those last without `JDataView` in node 6.x

```
$ node --version
v6.10.0
$ node
> const b = new DataView(new ArrayBuffer(3));
undefined
> b
DataView {
  byteLength: 3,
  byteOffset: 0,
  buffer: ArrayBuffer { byteLength: 3 } }
>  
```

**Comment by RReverser** [comment]
*sigh* This is why I asked to provide a demo. https://jsbin.com/kuyofimore/edit?html,console

TL;DR - jDataView supports ArrayBuffer just fine, the issue occurs because you're modifying built-in globals. Many things can go wrong when doing that in JavaScript. Just import the module under a separate name and it should work fine.

**Comment by greggman** [comment]
maybe you should put that in big giant letters because others are unaware
of the issues. I'm here because other multi module libraries are using
jdataview this way and digging into them this is the reason they were
failing.

They wrote the libraries to work with `DataView` and then dropped in your
library to support those platforms that don't have `DataView`

**Comment by RReverser** [comment]
It's a generic JavaScript rule of thumb, nothing specific to this library to justify explaining these things.

However, if you want to improve feature detection to avoid this case, I'll accept a PR.
